test/integration/supervisor_tls_spec_test.go

```diff
@@ -0,0 +1,296 @@
+			customResourceYaml: here.Doc(`
+			---
+			apiVersion: idp.supervisor.%s/v1alpha1
+			kind: OIDCIdentityProvider
+			metadata:
+				name: %s
+			spec:
+				tls:
+					certificateAuthorityDataSource:
+						kind: Secret
+						name: foo
+						key: bar
+				issuer: https://foo.bar.com/oauth2/default
+				authorizationConfig:
+					additionalScopes: [offline_access, email]
+					allowPasswordGrant: true
+				client:
+					secretName: foo-bar-client-credentials
+			`),
+			customResourceName: "valid-oidc-idp-secret-kind",
+			expectedError:      "",
+		},
+		{
+			name: "should create a custom resource passing all validations using a ConfigMap source",
+			customResourceYaml: here.Doc(`
+			---
+			apiVersion: idp.supervisor.%s/v1alpha1
+			kind: OIDCIdentityProvider
+			metadata:
+				name: %s
+			spec:
+				tls:
+					certificateAuthorityDataSource:
+						kind: ConfigMap
+						name: foo
+						key: bar
+				issuer: https://foo.bar.com/oauth2/default
+				authorizationConfig:
+					additionalScopes: [offline_access, email]
+					allowPasswordGrant: true
+				client:
+					secretName: foo-bar-client-credentials
+			`),
+			customResourceName: "valid-oidc-idp-cm-kind",
+			expectedError:      "",
+		},
+		{
+			name: "should create a custom resource without any tls spec",
+			customResourceYaml: here.Doc(`
+			---
+			apiVersion: idp.supervisor.%s/v1alpha1
+			kind: OIDCIdentityProvider
+			metadata:
+				name: %s
+			spec:
+				issuer: https://foo.bar.com/oauth2/default
+				authorizationConfig:
+					additionalScopes: [offline_access, email]
+					allowPasswordGrant: true
+				client:
+					secretName: foo-bar-client-credentials
+			`),
+			customResourceName: "no-tls-spec",
+			expectedError:      "",
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+			yamlFilepath := filepath.Join(t.TempDir(), fmt.Sprintf("tls-spec-validation-%s.yaml", tc.customResourceName))
+
+			resourceName := tc.customResourceName + "-" + testlib.RandHex(t, 7)
+			yamlBytes := []byte(fmt.Sprintf(tc.customResourceYaml, env.APIGroupSuffix, resourceName))
+
+			require.NoError(t, os.WriteFile(yamlFilepath, yamlBytes, 0600))
+			cmd := exec.CommandContext(context.Background(), "kubectl", []string{"apply", "-f", yamlFilepath}...)
+			var stdOut, stdErr bytes.Buffer
+			cmd.Stdout = &stdOut
+			cmd.Stderr = &stdErr
+			err := cmd.Run()
+			t.Cleanup(func() {
+				t.Helper()
+				require.NoError(t, exec.Command("kubectl", []string{"delete", "--ignore-not-found", "-f", yamlFilepath}...).Run())
+			})
+			if tc.expectedError == "" {
+				require.NoError(t, err)
+				require.Equal(t, fmt.Sprintf("oidcidentityprovider.idp.supervisor.pinniped.dev/%s created\n", resourceName), stdOut.String())
+				require.Empty(t, stdErr.String())
+				t.Cleanup(func() {
+					t.Helper()
+					require.NoError(t, exec.Command("kubectl", []string{"delete", "-f", yamlFilepath}...).Run())
+				})
```

Comment:
Looks like you meant to delete these lines.

Reply:
Yes, meant to delete this. Fixed now.